### PR TITLE
Patches to mruby.go to implement module lookup

### PR DESCRIPTION
**Pull request by erikh**
Hi folks, I hope this helps. Here's a more complete example than the tests needed based on the initial example in the README:

```go
package main

import (
  "fmt"

  mruby "github.com/mitchellh/go-mruby"
)

func main() {
  mrb := mruby.NewMrb()
  defer mrb.Close()

  // Our custom function we'll expose to Ruby. The first return
  // value is what to return from the func and the second is an
  // exception to raise (if any).
  addFunc := func(m *mruby.Mrb, self *mruby.MrbValue) (mruby.Value, mruby.Value) {
    args := m.GetArgs()
    return mruby.Int(args[0].Fixnum() + args[1].Fixnum()), nil
  }

  // Lets add a method to Kernel
  class := mrb.Module("Kernel")
  class.DefineClassMethod("add", addFunc, mruby.ArgsReq(2))

  // Let's call it and inspect the result
  result, err := mrb.LoadString(`Kernel.add(10, 15)`)
  if err != nil {
    panic(err.Error())
  }

  // This will output "Result: 42"
  fmt.Printf("Result: %s\n", result.String())
}
```

### CODE REVIEW

File: mruby_test.go [hunk 1]
```diff
@@ -19,6 +19,16 @@ func TestMrbArena(t *testing.T) {
...
 }
 
+func TestMrbModule(t *testing.T) {
+	mrb := NewMrb()
+	defer mrb.Close()
+
+	module := mrb.Module("Kernel")
+	if module == nil {
+		t.Fatal("module was nil and should not be")
+	}
```
Comment (mitchellh) [inline mruby_test.go]: Can you also add a negative test? (when a module that doesn't exist is requested)

**Comment by mitchellh** [comment]
Looks great, one request and we can move forward.

**Comment by erikh** [comment]
Ah sure, sorry about that!

**Comment by mitchellh** [comment]
Ah, hah. Well that is a frightening API. o.O For now let's just leave it as is.

**Pull request merged**